samples/EchoBot/EchoBot.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Middleware;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Samples.Echo
+{
+    public class EchoState : StoreItem
+    {
+        public string eTag { get; set; }
+
+        public int TurnNumber { get; set; }
+    }
+
+    public class EchoBot : IBot
```

Comment:
I really like how this divorces the bot from the underlying adapter. I also like how you drove this home by using the bot in multiple samples across multiple adapters.

Reply:
👍